dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -673,14 +682,11 @@ enum LastAction {
         this.currentMinimumPoolSize = this.coreThreads;
         this.maxThreads = pool.getMaximumPoolSize();
         this.threadRange = this.maxThreads - this.coreThreads;
-        setPoolSize(coreThreads);
+        setPoolSize(Math.min(startupPoolSize, this.maxThreads));
         targetPoolSize = coreThreads;
         resetStatistics(true);
-        // nothing to do if core == max
-        if (coreThreads < maxThreads) {
-            activeTask = new IntervalTask(this);
-            timer.schedule(activeTask, interval, interval);
-        }
+        activeTask = new IntervalTask(this);
```

Comment:
I'm not sure I understand the question. 

When we only had `coreThreads` and `maxThreads`, if they were equal then the controller had nothing to do, so it did not need to wake up every 1500 ms to check the pool stats, so we did not set the timer.

Now that we are adding `startupPoolSize` to the mix, we need the controller to do the periodic checks until "ready to run" so that it can see the notification that startup is complete and switch from `startupPoolSize` to `coreThreads`. 

Those checks are extremely lightweight, just a couple of if statements in `evaluateInterval` (shown later in the changeset, starting on line 1344). But we have to do this, or the pool will stay in `startupPoolSize` forever ...

Reply:
My question was for the case when `startupPoolSize == coreThreads == maxThreads`. The logic for not running the controller in that case is similar to why we would not have run it originally if `coreThreads == maxThreads`

Reply:
The "overhead" I was asking about was for the controller doing its logic every 1.5 seconds on an ongoing basis even when there is nothing to do. If that overhead is negligible, then I agree there is (and never was) any point in avoiding running that logic (even originally when `coreThreads` == `maxThreads`)

Reply:
Ah-ha, now I understand the question, thanks. 

And yes, we could get a small optimization from not activating the timer in this case:
`startupPoolSize == coreThreads == maxThreads`

I can add that in there, because in that case there is _still_ nothing for the controller to do, so no point in it waking up every 1500 ms. The overhead is very small, but it will still add a bit to idle cpu time.

Reply:
Pushed a commit with the change to avoid running the controller cycle if `startupPoolSize == coreThreads == maxThreads`

Reply:
Thanks, yes this is what I had in mind with my question.